tests/integrations/raftstore/test_hibernate.rs

```diff
@@ -275,3 +276,137 @@ fn test_split_delay() {
     cluster.clear_send_filters();
     must_get_equal(&cluster.get_engine(4), b"k2", b"v2");
 }
+
+/// During rolling update, hibernate configuration may vary in different nodes.
+/// If leader sleep unconditionally, follower may start new election and cause
+/// jitters in service. This case tests leader will go to sleep only when every
+/// followers are agreed to.
+#[test]
+fn test_inconsistent_configuration() {
+    let mut cluster = new_node_cluster(0, 3);
+    configure_for_hibernate(&mut cluster);
+    cluster.run();
+    cluster.must_transfer_leader(1, new_peer(1, 1));
+    cluster.must_put(b"k1", b"v1");
+    must_get_equal(&cluster.get_engine(3), b"k1", b"v1");
+
+    // Wait till leader peer goes to sleep.
+    thread::sleep(
+        cluster.cfg.raft_store.raft_base_tick_interval.0
+            * 3
+            * cluster.cfg.raft_store.raft_election_timeout_ticks as u32,
+    );
+
+    // Ensure leader can sleep if all nodes enable hibernate region.
+    let awakened = Arc::new(AtomicBool::new(false));
+    let filter = Arc::new(AtomicBool::new(true));
+    let a = awakened.clone();
+    cluster.add_send_filter(CloneFilterFactory(
+        RegionPacketFilter::new(1, 1)
+            .direction(Direction::Send)
+            .set_msg_callback(Arc::new(move |_| {
+                a.store(true, Ordering::SeqCst);
+            }))
+            .when(filter.clone()),
+    ));
+    thread::sleep(cluster.cfg.raft_store.raft_heartbeat_interval() * 2);
```

Comment:
Should we sleep more than 2 to make the test more stable?

Reply:
We only need to sleep a heartbeat timeout to receive a heartbeat, using `* 2` is already considering the occasional slowness in tests.